### DISCUSSION TITLE
Make httpstatuses.com noise in exceptions optional

### DISCUSSION THREAD

**Discussion by ionelmc** [encode/httpx#2766] [General] [answered]
To me this only looks like unnecessary and annoying noise: 

> For more information check: https://httpstatuses.com/123

That domain redirects to some company website full of advertisements and no information for anything but a handful of status codes.

I'm assuming there's some sponsorship for this project coming from that company, and that is ok. But please, have a way to opt-out so people don't have cleanup the exception messages.

**Comment by lovelydinosaur** [comment, marked as answer]
Oh fun. Looks like that site has become shit.

> I'm assuming there's some sponsorship for this project coming from that company

Nope. It used to be a really nice resource. It's not now.

Let's have a design discussion about what we'd like to change the messaging too, and get it done.

We could...

* Switch over to something like Mozilla's docs, eg... "https://developer.mozilla.org/en-US/docs/Web/HTTP/Status/303"
* Drop the link altogether.

**Reply by ionelmc** [reply]
The Mozilla one is good but it's a lot of text.

I think (or hope) most users of this library don't really need a link to a website to figure out what a status code is about.

I'd vote for removing.

**Reply by lovelydinosaur** [reply]
Alt: https://http.cat/status/303 😼

**Reply by zanieb** [reply]
I think including a link for more information is nice. It's good practice for an error message to direct to additional resources. I agree it looks like httpstatuses.com has gone downhill :(

**Reply by lovelydinosaur** [reply]
Okay, I'm agnostic but *either* of those options would be far better than what we're pointing to at the moment.
Let's okay whichever pull request comes in for this.

**Reply by guy-ecton** [reply]
+1 for removing this error (or at least being able to turn it off).
The stacktrace is already long enough, and I would like to see the error quickly, and not read the entire URL for mozilla docs...
